WordPress/Classes/Stores/StatsPeriodStore.swift

```diff
@@ -1373,5 +1379,94 @@ private extension PeriodStoreState {
+               let blog = blogService.blog(byBlogId: blogID) {
+                // set the title to the site title, if it's not nil and not empty; otherwise use the site url
+                let title = (element.title ?? url).isEmpty ? url : element.title ?? url
+                let timeZone = blogService.timeZone(for: blog)
+
+                result[blogID.intValue] = HomeWidgetThisWeekData(siteID: blogID.intValue,
+                                                                 siteName: title,
+                                                                 url: url,
+                                                                 timeZone: timeZone,
+                                                                 date: Date(timeIntervalSinceReferenceDate: 0),
+                                                                 stats: ThisWeekWidgetStats())
+
+            }
+        }
+    }
+
+    func storeThisWeekHomeWidgetData(stats: ThisWeekWidgetStats) {
+        guard #available(iOS 14.0, *),
+              let siteID = SiteStatsInformation.sharedInstance.siteID else {
+            return
+        }
+
+        var homeWidgetCache = HomeWidgetThisWeekData.read() ?? initializeThisWeekWidgetData()
+        guard let oldData = homeWidgetCache[siteID.intValue] else {
+            DDLogError("This Week Widget: Failed to find a matching site")
+            return
+        }
+        let blogService = BlogService(managedObjectContext: ContextManager.shared.mainContext)
+
+        guard let blog = blogService.blog(byBlogId: siteID) else {
+            DDLogError("StatsWidgets: the site does not exist anymore")
+            // if for any reason that site does not exist anymore, remove it from the cache.
+            homeWidgetCache.removeValue(forKey: siteID.intValue)
+            HomeWidgetThisWeekData.write(items: homeWidgetCache)
+            return
+        }
+        homeWidgetCache[siteID.intValue] = HomeWidgetThisWeekData(siteID: siteID.intValue,
+                                                                  siteName: blog.title ?? oldData.siteName,
+                                                                  url: blog.url ?? oldData.url,
+                                                                  timeZone: blogService.timeZone(for: blog),
+                                                                  date: Date(),
+                                                                  stats: stats)
+        HomeWidgetThisWeekData.write(items: homeWidgetCache)
+        WidgetCenter.shared.reloadTimelines(ofKind: WPHomeWidgetThisWeekKind)
+    }
+
+    func updateThisWeekHomeWidget() {
+        guard #available(iOS 14.0, *) else {
+            return
+        }
+        switch summary?.period {
+        case .day:
+            guard summary?.periodEndDate == StatsDataHelper.currentDateForSite().normalizedDate() else {
+                WidgetCenter.shared.reloadTimelines(ofKind: WPHomeWidgetThisWeekKind)
```

Comment:
What's the purpose of reloading the timelines here and in the `.week` case below? Do we not just want to populate the data if we have new data and ignore it otherwise? Otherwise, based on my testing, we're calling reloadTimelines every time the user accesses any week page in the Stats section.

Reply:
Agreed, we should review this once we merge [this](https://github.com/wordpress-mobile/WordPress-iOS/pull/15838)